egs/swbd/s5c/local/nnet3/run_tdnn.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# this is the standard "tdnn" system, built in nnet3; it's what we use to
+# call multi-splice.
+
+. cmd.sh
+
+
+# At this script level we don't support not running on GPU, as it would be painfully slow.
+# If you want to run without GPU you'd have to call train_tdnn.sh with --gpu false,
+# --num-threads 16 and --minibatch-size 128.
+
+stage=0
+train_stage=-10
+has_fisher=true
+speed_perturb=true
+dir=exp/nnet3/nnet_tdnn_a
+train_set=train_nodup
+ali_dir=exp/tri4_ali_nodup
+if [ "$speed_perturb" == "true" ]; then
+  dir=${dir}_sp
+  train_set=train_nodup_sp
+  ali_dir=exp/tri4_ali_nodup_sp
+fi
+
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+local/nnet3/run_ivector_common.sh --stage $stage || exit 1;
```

Comment:
Please pass on the option for speed perturbation to the run_ivector_common.sh script. You could do something like 

<pre><code>
suffix=
if [ "$speed_perturb" == "true" ]; then
  suffix=_sp
fi
dir=exp/nnet3/tdnn
dir=$dir${affix:+_$affix}
dir=${dir}$suffix
train_set=train_nodup$suffix
ali_dir=exp/tri4_ali_nodup$suffix

local/nnet3/run_ivector_common.sh --stage $stage \
  --speed-perturb $speed_perturb || exit 1;
</code></pre>